### ISSUE TITLE
Rubocop Flagging Style/FormatStringToken on CI only

### ISSUE THREAD

**Issue by luke-hill** [rubocop-hq/rubocop#5483]
## Expected behavior

Rubocop is a gem, and should be platform agnostic

## Actual behavior

Rubocop on our CI environment flags the following rule. Running this locally passes this check.

```
features/support/helpers/screenshots.rb:12:32: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
      Time.now.strftime('%Y-%m-%d-%H-%M-%S')
```

## Steps to reproduce the problem
Create this method. Try on CI (Jenkins) / locally
```
def timestamp
  Time.now.strftime('%Y-%m-%d-%H-%M-%S')
end
```

## RuboCop version
`0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.3 i386-mingw32)`

I expect there to be some issues reproducing this. I'd rather not add this to the ignore list incase other issues crop up.

**Comment by pocke** [comment]
Hmm, I'm not sure why the difference occurs. In my environment, the cop does not add any offenses to the code.
And I think the cop is implemented that does not add any offenses. The cop checks only `sprintf`, `format` and `%` methods.

https://github.com/bbatsov/rubocop/blob/v0.52.1/lib/rubocop/cop/style/format_string_token.rb#L45
https://github.com/bbatsov/rubocop/blob/v0.52.1/lib/rubocop/cop/style/format_string_token.rb#L70-L77


But it looks having a bug. The cop adds a falsy offense  to the following code.

```ruby
def timestamp
  Time.now.strftime('%Y-%m-%d-%H-%M-%S')
  format
end
```

```bash
$ rubocop 
Inspecting 1 file
C

Offenses:

test.rb:2:28: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
  Time.now.strftime('%Y-%m-%d-%H-%M-%S')
                           ^^

1 file inspected, 1 offense detected
```


So, can you check if `features/support/helpers/screenshots.rb` has `format`, `sprintf`or `%` method call?

**Comment by luke-hill** [comment]
Few questions (As I'm still only around 9-12 months Ruby XP)

1) What exactly is this cop complaining about (In easy to understand language)
2) Our CI builds in Jenkins using Shell, but outputs to Windows boxes, and I'm on Windows, so I'm not sure it's a platform issue
3) For your question about checking `features/support/helpers/screenshots.rb` do you want me to check the whole file?

**Comment by pocke** [comment]
> 1. What exactly is this cop complaining about (In easy to understand language)

This cop checks format string, like `sprintf("%s", 'foo')`. In Ruby, we can use "named" and "typed" format string. 
For example:

```ruby
# unnamed, typed format. `%s` accepts a string.
sprintf("foo%s", "bar") # => "foobar"

# named, untyped format. `%{something}` is named, and it accepts anything.
sprintf("foo%{something}", something: "bar") # => "foobar"

# named, typed format. `%<something>s` is named, and it accepts a string.
sprintf("foo%<something>s", something: "bar") # => "foobar"
```

This cop prefer the last style (`%<something>s`).

But in this case, `%Y-%m-%d-%H-%M-%S` is not a format string for `sprintf`. It is for `strftime`. We can't use the named format for `strftime`, so the cop should not add any offenses to the code.

> 2. Our CI builds in Jenkins using Shell, but outputs to Windows boxes, and I'm on Windows, so I'm not sure it's a platform issue

I'm not sure also. Sometimes RuboCop has different behaviour from *nix in Windows. but i don't have Windows machine so I can't check it.
And I guess it is not a platform issue, since the cop does not seem to have platform specific code.

> 3. For your question about checking features/support/helpers/screenshots.rb do you want me to check the whole file?

Yes. This cop adds an offense to `strftime('%Y-%m-%d-%H-%M-%S')` by the bug if `format`, `sprintf` or `%` methods is somewhere in the file.
So, if the file has difference that is existence of these methods between your CI and local, it may be the cause.

And I'll fix the bug.

**Comment by luke-hill** [comment]
Entire file:

```
# frozen_string_literal: true

module Helpers
  module Screenshots
    def screenshot_name
      @screenshot_name ||= "#{tld}/screenshots/failure_#{timestamp}.png"
    end

    private

    def timestamp
      Time.now.strftime('%Y-%m-%d-%H-%M-%S')
    end

    def tld
      File.absolute_path('.')
    end
  end
end
```

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
